go/libraries/doltcore/sqle/dfunctions/dolt_reset.go

```diff
@@ -67,8 +67,9 @@ func (d DoltResetFunc) Eval(ctx *sql.Context, row sql.Row) (interface{}, error)
 		return 1, err
 	}
 
+	var h string
 	if apr.Contains(cli.HardResetParam) {
-		h, err := actions.ResetHardTables(ctx, dbData, apr, working, staged, head)
+		h, err = actions.ResetHardTables(ctx, dbData, apr, working, staged, head)
```

Comment:
Restore this line plz. h should be scoped to within the if block

Reply:
OK. I scoped it outside the block so that I could get the `err` that was being swallowed out of the block. I'll just add an error check inside.